pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainScheduler.java

```diff
@@ -55,4 +55,9 @@ public interface OpChainScheduler {
    *         prior to this call
    */
   OpChain next();
+
+  /**
+   * @return the number of operator chains that are awaiting execution
+   */
+  int size();
```

Comment:
this is a weak reference to the queue yes? e.g. no need for lock

Reply:
there's no need for lock because OCSchedulers are not thread safe - thread safety should come from caller if they care about it